rmw_fastrtps_shared_cpp/test/test_dds_qos_policy_to_rmw.cpp

```diff
@@ -0,0 +1,153 @@
+// Copyright 2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gtest/gtest.h"
+
+#include "rmw_fastrtps_shared_cpp/qos.hpp"
+#include "fastrtps/attributes/PublisherAttributes.h"
+#include "fastrtps/attributes/SubscriberAttributes.h"
+#include "fastrtps/qos/QosPolicies.h"
+#include "fastrtps/qos/WriterQos.h"
+#include "fastrtps/qos/ReaderQos.h"
+
+typedef eprosima::fastrtps::PublisherAttributes PublisherAttributes;
+typedef eprosima::fastrtps::SubscriberAttributes SubscriberAttributes;
+typedef eprosima::fastrtps::WriterQos WriterQos;
+typedef eprosima::fastrtps::ReaderQos ReaderQos;
+
+typedef eprosima::fastrtps::HistoryQosPolicy HistoryQosPolicy;
+typedef eprosima::fastrtps::TopicAttributes TopicAttributes;
+
+TEST(test_dds_qos_policy_to_rmw, default_publisher_attributes) {
```

Comment:
```
static constexpr const rmw_qos_profile_t EXPECTED_DEFAULT_QOS_PROFILE = {{
 .depth = 1,
 .history = ...,
 // etc.
}};
```

then you only need a single EQ statement.

Reply:
You probably want to read https://github.com/google/googletest/blob/master/googlemock/docs/cheat_sheet.md

Reply:
https://github.com/google/googletest/blob/master/googlemock/docs/cook_book.md